src/app/static/src/tests/components/adr/selectDataset.test.ts

```diff
@@ -528,6 +555,76 @@ describe("select dataset", () => {
         expect(rendered.find("#loading-dataset").exists()).toBe(false);
         expect(rendered.find(Modal).props("open")).toBe(false);
     });
+    
+    it("renders reset confirmation dialog when changing selected dataset and then modal if click save", async () => {
+        let store = getStore({
+            selectedDataset: fakeDataset
+        },
+            {datasets: [{...fakeRawDatasets[0], ...fakeRawDatasets[1], resources: [shape]}]}, true
+        )
+        const rendered = mount(SelectDataset, {
+            store, stubs: ["tree-select"]
+        });
+        rendered.find("button").trigger("click");
+
+        await Vue.nextTick();
```

Comment:
Please see whether `await`ing above enables you to remove this `Vue.nextTick()` - I didn't need any when I was playing with this.